File: schemas/booking-scheduling/drizzle/providers.ts
// providers: individuals who deliver bookable services.
// See README.md for full design rationale.

import { pgTable, uuid, text, boolean, integer, timestamp, index } from "drizzle-orm/pg-core";
import { users } from "./users";

export const providers = pgTable(
  "providers",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    userId: uuid("user_id").unique().notNull().references(() => users.id, { onDelete: "cascade" }),
    displayName: text("display_name").notNull(),
    bio: text("bio"),
    avatarUrl: text("avatar_url"),
    timezone: text("timezone").notNull(),
    phone: text("phone"),
    email: text("email"),
    isActive: boolean("is_active").notNull().default(true),
    isAccepting: boolean("is_accepting").notNull().default(true),
    position: integer("position").notNull().default(0),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_providers_is_active_is_accepting").on(table.isActive, table.isAccepting),
    index("idx_providers_is_active_position").on(table.isActive, table.position),
  ]
);

File: schemas/booking-scheduling/spacetimedb/providers.rs
// providers: Service providers who can be booked for appointments.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = providers, public)]
pub struct Provider {
    #[primary_key]
    pub id: String, // UUID
    #[unique]
    pub user_id: String, // UUID — FK → users.id (cascade delete)
    pub display_name: String,
    pub bio: Option<String>,
    pub avatar_url: Option<String>,
    pub timezone: String,
    pub phone: Option<String>,
    pub email: Option<String>,
    #[index(btree)]
    pub is_active: bool,
    pub is_accepting: bool,
    pub position: i32,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
    // Composite index: (is_active, is_accepting)
    // Composite index: (is_active, position)
}
